bot-components/GitHub_mutations.ml

```diff
@@ -150,11 +150,11 @@ let create_check_run ~bot_info ?conclusion ~name ~repo_id ~head_sha ~status
   |> serializeVariables |> variablesToJson
   |> GraphQL_query.send_graphql_query ~bot_info ~query
        ~parse:(Fn.compose parse unsafe_fromJson)
-  >|= function
-  | Ok _ ->
-      ()
-  | Error err ->
-      Stdio.print_endline (f "Error while creating check run: %s" err)
+  >|= Result.bind ~f:(function
+        | {createCheckRun= Some {checkRun= Some {url}}} ->
+            Ok url
+        | _ ->
+            Error (f "Error while creating check run.") )
```

Comment:
You should keep propagating the error message instead of dropping it. This would give you detailed information on how you failed to get the check run.

Reply:
I think I copied the comment case a bit too literally here. I will try to make a more specific match and propagate the error correctly like you said.

Reply:
Wait isn't this exactly what Result.bind is doing. Here we are separating the case where there is None for the url field and returning an error. The original error would be preserved.

Reply:
Very well, I will remove the link.

Reply:
Or maybe just keep the code so that if GitHub starts being more reactive and returning the check run URL we can see it appear in our PRs...

Reply:
OK I won't touch it then. I set it up so that it doesn't impede the main function anyway.

Reply:
Maybe just replace the "Error" by a "Warning" so that we do not get error messages in the log for something that we already know is not working.